tests/sentry/incidents/endpoints/test_serializers.py

```diff
@@ -83,3 +85,33 @@ def test_aggregations(self):
             {'aggregations': [50]},
             {'aggregations': invalid_values},
         )
+
+    def _run_changed_fields_test(self, alert_rule, params, expected):
+        serializer = AlertRuleSerializer(
+            context={'project': self.project},
+            instance=alert_rule,
+            data=params,
+            partial=True,
+        )
+        serializer.is_valid()
+        assert serializer._remove_unchanged_fields(
```

Comment:
Could these permutations be tested via an endpoint test instead of testing the internal methods directly? This seems like it might be more fragile than an endpoint test.

Reply:
Not sure that I want to test all the permutations of this in endpoint tests, this seems like a better place for all the specifics. I've added `test_not_updated_fields` to the endpoint tests just to make sure it works as expected in at least one case.